kart/tabular/table_dataset.py

```diff
@@ -134,25 +132,16 @@ def features(self, spatial_filter=SpatialFilter.MATCH_ALL, log_progress=False):
         spatial_filter - restricts the features yielded to those that are in a particular geographic area.
         log_progress - can be set to True, or to a callable logger method eg L.info, to enable logging.
         """
-        if log_progress:
-            plog = L.info if log_progress is True else log_progress
-            log_progress = bool(log_progress)
-
         spatial_filter = spatial_filter.transform_for_dataset(self)
 
         n_read = 0
-        n_chunk = 0
         n_matched = 0
-        n_total = self.feature_count
-        t0 = time.monotonic()
-        t0_chunk = t0
-
         if log_progress:
-            plog("0.0%% 0/%d features... @0.0s", n_total)
+            n_total = self.feature_count
+            progress = tqdm(total=n_total, unit="F", desc=self.path)
```

Comment:
worth using it as a contextmanager instead? makes it clearer that `close()` will be called. and fallback to `contextlib.null_context()` when `not log_progress`

Reply:
Done